### ISSUE TITLE
Fixing in "PEAR.WhiteSpace.ObjectOperatorIndent" sniff results in infinite loop

### ISSUE THREAD

**Issue by aik099** [squizlabs/PHP_CodeSniffer#406]
Example code:
```php
$response -> CompletedTrackDetails -> TrackDetails -> Events -> Address -> CountryCode
```

Relevant part from `ruleset.xml`:
```
<rule ref="PEAR.WhiteSpace.ObjectOperatorIndent">
	<properties>
		<property name="indent" value="1"/>
	</properties>
</rule>
```

I expected new line to be placed before each `->` and then indentation added. Instead the spaces are added before/after the `->` until 50 fixing passes are reached.

**Comment by gsherwood** [comment]
That code gives me no errors when I set the indent value to `1`, and nothing is fixed when I run PHPCBF.

That sniff doesn't check that each object operator is on a new line, which is what I think you want. This isn't the right sniff for you. I don't think there is one that does this.

I do think there are some problems with the sniff though as it is reporting errors for those object operators when it should be ignoring them as they are not at the start of a line.

**Comment by gsherwood** [comment]
I've fixed the sniff so it only checks object operators at the start of the line, as it is supposed to (https://github.com/squizlabs/PHP_CodeSniffer/commit/c8a7e92af24463ae82d8136a937754c82898d031). It doesn't (and shouldn't) enforce that they must be on new lines. That's not its job.

Do you still get issues with this change in place?

**Comment by aik099** [comment]
> That sniff doesn't check that each object operator is on a new line, which is what I think you want. This isn't the right sniff for you. I don't think there is one that does this.

Either it's a bug in a sniff or it's documentation, because example in documentation says that: https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/PEAR/Docs/WhiteSpace/ObjectOperatorIndentStandard.xml#L26-L28

> Do you still get issues with this change in place?

Sadly I then realized that sniff is absolutely not what I needed (according to example in docs) and I stopped using it. But thanks for fixing anyway. Theoretically adding it back to https://github.com/aik099/CodingStandard/ should no longer result in infinite fixing.

**Comment by gsherwood** [comment]
> Either it's a bug in a sniff or it's documentation

It checks the indent of object operators *if and only if* you already have them on new lines. It doesn't enforce the new line otherwise you could never write `$foo->bar()->baz();` on one line.

So yes as you've discovered, it's probably not the sniff you are looking for.

**Comment by midwestfishingadventures** [comment]
hey I am have a problem with this sniff. Or at least I think I am. My code:
$result = $db -> query($sql, $params);
object operator not indented correctly. Expected 8 found 1:

**Comment by aik099** [comment]
I think that's another sniff. You'd better open another task and `phpcs -s` would show you the sniff that reports the error.
